UI DreamServer Upgrade

SergioDiazF opened the pull request:
Hello! first time using Github, so hope i'm doing things well. I've created a new UI graphic version of "DreamServer" site, so hope you like it. 

I mainly added some tags to reorder the base and reused the IDs and Classes already present to give them a new style. It is Responsive. Anything new you need I'm at your disposal.


![Example](https://user-images.githubusercontent.com/69328998/186976527-c51ea1a7-2ff2-4020-9ed1-e224edde7e0b.jpg)

namion commented:
I like this. I am new to github also, so I don't know where to report these things, but when using img2img, there is nowhere in the UI to adjust the strength parameter. I reported this issue on another comment chain related to the original web UI and figured I would mention it here as well.

TesseractCat commented:
Would it be possible to refactor this PR to create a more generic system for applying themes? Thus keeping both the original theme, this theme, and add the possibility for different themes in the future.

TesseractCat commented on static/dream_web/index.css:
```diff
@@ -1,61 +1,274 @@
-* {
-    font-family: 'Arial';
+@import url('https://fonts.googleapis.com/css2?family=Roboto:wght@400;500;700;900&display=swap');
+:root{
```
There should be a space between the selector and the opening brackets. See: https://google.github.io/styleguide/htmlcssguide.html

SergioDiazF replied:
Solved

TesseractCat commented on static/dream_web/index.css:
```diff
@@ -1,61 +1,274 @@
+}
+*/
+
+/*---------------------BOTON-SUBMIT---------------------*/
```
*BUTTON

SergioDiazF replied:
Solved

TesseractCat commented on static/dream_web/index.css:
```diff
@@ -1,61 +1,274 @@
-    width: 150px;
+
+
+/*----------Responsive----------*/
```
Should be all caps to keep with other comments

SergioDiazF replied:
Solved

TesseractCat commented on static/dream_web/index.css:
```diff
@@ -1,61 +1,274 @@
+#seed{
+    width: 110px;
+}
+/*
```
Should this commented section remain here?

SergioDiazF replied:
Solved.

TesseractCat commented on static/dream_web/index.html:
```diff
@@ -8,58 +8,73 @@
+                      <option value="960">960</option> <option value="1024">1024</option>
+                    </select>
+                </div>
+                <div class="config-container">
```
Why was the reset button removed?

SergioDiazF replied:
Solved.

TesseractCat commented on static/dream_web/index.html:
```diff
@@ -8,58 +8,73 @@
+                      <option value="64">64</option> <option value="128">128</option>
+                      <option value="192">192</option> <option value="256">256</option>
+                      <option value="320">320</option> <option value="384">384</option>
+                      <option value="448">448</option> <option value="512" selected="selected">512</option>
```
Since the default value is provided by the `value` attribute of the select element, there is no need to add a `selected` attribute here.

SergioDiazF replied:
I think the 512 size is more likely to be used, so I added the `selected` attribute to be the default value.

TesseractCat reviewed (changes requested):
Added some quick comments. Also the background image should probably be a compressed JPG, for space reasons.

SergioDiazF commented:
> Added some quick comments. Also the background image should probably be a compressed JPG, for space reasons.

Thank u! Totally agree with your comments, i'm working to solve them.

lstein commented:
@SergioDiazF This is very attractive. I particularly like the generous resizable prompt box. It's not clear to me whether the "various updates" commit you made address @TesseractCat's comments? I'll wait for confirmation

TesseractCat commented:
Hi, my main concern is this:

> Would it be possible to refactor this PR to create a more generic system for applying themes? Thus keeping both the original theme, this theme, and add the possibility for different themes in the future.

Personally I prefer the old style, so I'd like it to be configurable. With a configurable system other users could contribute styles as well.

bakkot commented:
I think having a fully configurable UI is probably too much complexity to be worth it, but most of these changes are just CSS, so replacing the new `index.css` with the old would probably get you the old UI back.

If we really do want it to be configurable, at least in part, how about having the server take a `--css-path` option?

(That said, I also prefer not having the background image - it's pretty busy, and distracts from the generated images, which after all are the point.)

TesseractCat commented:
> I think having a fully configurable UI is probably too much complexity to be worth it, but most of these changes are just CSS, so replacing the new `index.css` with the old would probably get you the old UI back.
> 
> If we really do want it to be configurable, at least in part, how about having the server take a `--css-path` option?
> 
> (That said, I also prefer not having the background image - it's pretty busy, and distracts from the generated images, which after all are the point.)

Yeah, all I mean by configurable is user-selectable css file.

SergioDiazF commented:
> @SergioDiazF This is very attractive. I particularly like the generous resizable prompt box. It's not clear to me whether the "various updates" commit you made address @TesseractCat's comments? I'll wait for confirmation

I took into account the recommendations of  @TesseractCat, so:

- Changed background.png to a light background.jpg image.
- added a space between all selectors and their respective brackets in css file.
- Changed all CSS comments to Caps.
- Cleaned CSS file removing useless commented code.
- Restore Seed reset button.

SergioDiazF commented:
> > I think having a fully configurable UI is probably too much complexity to be worth it, but most of these changes are just CSS, so replacing the new `index.css` with the old would probably get you the old UI back.
> > If we really do want it to be configurable, at least in part, how about having the server take a `--css-path` option?
> > (That said, I also prefer not having the background image - it's pretty busy, and distracts from the generated images, which after all are the point.)
> 
> Yeah, all I mean by configurable is user-selectable css file.

Totally agree, although the UI design has as main objective to facilitate the use of the tool, it also has a personal touch in between, so integrating different creative looks will cover a larger group of people. And it's not that difficult to implement considering that you only have to modify a `CSS` file.

LoopControl commented on static/dream_web/index.css:
```diff
@@ -1,61 +1,268 @@
+@import 'https://fonts.googleapis.com/css2?family=Roboto:wght@400;500;700;900&display=swap';
```
Why is this `import` here if the "Roboto" font isn't even used in the CSS?

Your `body {}` has a `font-family: 'Montserrat', sans-serif;`. I'd change the `body` CSS to have `font-family: 'Roboto', Arial, Helvetica, sans-serif; ` if you want to use Roboto.

Also remove the import from `fonts.googleapis.com`. We don't need to send a request to Google servers when we load the local UI  just because we're missing a non-essential font. It also will fail if the users are using SD offline.

lstein replied:
I agree with this comment. I'm personally using a GPU that is in an HPC cluster and can't access the internet, and have been working hard to work around all the other places where stable diffusion tries to download model or parameter files.

lstein replied:
@SergioDiazF Just checking in. Are you planning to address the change requests? I'll keep the PR open until you confirm one way or another.

SergioDiazF replied:
I confirm the change, I fully agree with the needs.

SergioDiazF commented:
Added theme support. To add new themes just include `CSS` file into /theme folder and add a new `<option value="theme_n" selected>Theme name</option>` inside `<select id="theme-selector">`, where "theme_n" is the name of the `CSS` file. 

![theme_1](https://user-images.githubusercontent.com/69328998/187309587-73202996-2d14-45d9-a0bb-4609d446cd74.jpg)
![theme_2](https://user-images.githubusercontent.com/69328998/187309589-c56353da-2594-40cc-80c6-46d6af342ea8.jpg)

Chase-san commented:
The background and theme systems seem unneeded. Seem like changes for the sake of changes.

lstein commented:
@SergioDiazF Unfortunately there have been a lot of changes to the WebGUI to add new features over the past couple of days, and my attempts to resolve the conflicts in index.html were not particularly successful. I would really like to see the theming go in, and if you have the time and energy for it, could you pull a fresh copy of main and make your changes against that? Also, if at all possible, try to end your lines with LF rather than CRLF? I believe Notepad++ lets you do this.

So sorry. I should have merged earlier, but there were lots of backed up PRs.

SergioDiazF commented:
> @SergioDiazF Unfortunately there have been a lot of changes to the WebGUI to add new features over the past couple of days, and my attempts to resolve the conflicts in index.html were not particularly successful. I would really like to see the theming go in, and if you have the time and energy for it, could you pull a fresh copy of main and make your changes against that? Also, if at all possible, try to end your lines with LF rather than CRLF? I believe Notepad++ lets you do this.
> 
> So sorry. I should have merged earlier, but there were lots of backed up PRs.

I completely undestand, So i need to close the pull request right? how can i do that?

lstein commented:
@SergioDiazF I hope you're still up for adding the theme changing features? I've caught up with the backlog of feature additions, and I think the web interface will be stable now.
